### PR TITLE
Various fixes to game-side code that are needed for w3dviewer

### PR DESCRIPTION
**Pull request by jonwil**

### CODE REVIEW

**Review by tomsons26 — APPROVED** [review]

File: src/w3d/lib/hashtemplate.h [hunk 1]
```diff
@@ -213,20 +216,23 @@ template<typename Key, typename Value> class HashTemplateIterator
...
-        int size = m_hashTable->m_size;
 
-        for (m_hashIndex++; m_hashIndex < size; m_hashIndex++) {
-            m_handle = m_hashTable->m_hash[m_hashIndex];
+        if (m_handle == -1) {
+            int size = m_hashTable->m_size;
 
-            if (m_handle != -1) {
-                break;
+            for (m_hashIndex++; m_hashIndex < size; m_hashIndex++) {
```
Comment (xezon) [inline src/w3d/lib/hashtemplate.h:223]: Is `for (m_hashIndex++;` correct? It looks very strange.

File: src/w3d/lib/threadtrack.cpp [hunk 2]
```diff
@@ -15,6 +15,11 @@
 #include "threadtrack.h"
 #include <cstring>
 
+#pragma warning(push)
+#pragma warning(disable : 4073) // warning C4073: initializers put in library initialization area
+#pragma init_seg(lib) // Forces objects and variables in this file to initialize before other stuff.
```
Comment (xezon) [inline src/w3d/lib/threadtrack.cpp:20]: Why is this necessary? What happens without this?
